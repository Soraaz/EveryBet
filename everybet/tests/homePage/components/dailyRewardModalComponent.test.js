import React from "react";
import renderer from "react-test-renderer";
import {shallow} from "enzyme";

import DailyRewardModalComponent from "../../../src/homePage/components/dailyRewardModalComponent";
import Button from "../../../src/common/components/button";
import User from "../../../src/classes/user";

describe("DailyRewardModalComponent", () => {
    let closeModal;

    beforeEach(() => {
        User.rewardValue = 0;
        closeModal = sinon.spy();
    });

    describe("rendering", () => {
        it("Should render correctly the component", () => {
            const tree = renderer
                .create(
                    <DailyRewardModalComponent
                        showModal={true}
                        closeModal={closeModal}
                    />
                )
                .toJSON();
            expect(tree).to.matchSnapshot();
        });

        it("Should render correctly the component (User.rewardTier > 1)", () => {
            User.rewardTier = 2;
            const tree = renderer
                .create(
                    <DailyRewardModalComponent
                        showModal={true}
                        closeModal={closeModal}
                    />
                )
                .toJSON();
            expect(tree).to.matchSnapshot();
        });
    });

    describe("functionalities", () => {
        let wrapper;

        beforeEach(() => {
            wrapper = shallow(
                <DailyRewardModalComponent
                    showModal={true}
                    closeModal={closeModal}
                />
            );
        });

        it("Should call closeModal when pressing the \"Merci !\" button", () => {
            expect(closeModal).to.be.not.called;
            wrapper.find(Button).first().props().onPress();
            expect(closeModal).to.be.calledOnce;
        });
    });
});
