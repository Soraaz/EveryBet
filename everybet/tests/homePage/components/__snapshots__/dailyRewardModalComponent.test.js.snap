// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DailyRewardModalComponent rendering Should render correctly the component (User.rewardTier > 1) 1`] = `
<Modal
  hardwareAccelerated={false}
  onRequestClose={[Function]}
  transparent={true}
  visible={true}
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#00000040",
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#ffffff",
            "borderRadius": 5,
            "width": "70%",
          },
          Object {
            "paddingBottom": 10,
          },
        ]
      }
    >
      <Text
        style={
          Object {
            "fontSize": 15,
            "fontWeight": "bold",
            "marginTop": 10,
            "textAlign": "center",
          }
        }
      >
        Récompense quotidienne
      </Text>
      <View
        style={
          Object {
            "borderBottomColor": "#000000",
            "borderBottomWidth": 1,
            "marginVertical": 10,
            "width": "50%",
          }
        }
      />
      <Text
        style={
          Object {
            "fontSize": 12,
            "fontWeight": "bold",
            "marginHorizontal": 10,
            "textAlign": "center",
          }
        }
      >
        Vous vous êtes connecté à EveryBet 2 jours d'affilé !
      </Text>
      <Text
        style={
          Object {
            "fontSize": 12,
            "fontWeight": "bold",
            "marginHorizontal": 10,
            "textAlign": "center",
          }
        }
      >
        Vous avez gagné 0
        <Image
          source={
            Object {
              "testUri": "../../../assets/images/betcoin_icon.png",
            }
          }
          style={
            Object {
              "height": 12,
              "width": 12,
            }
          }
        />
         comme cadeau pour votre fidélité.
      </Text>
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignSelf": "center",
            "backgroundColor": "#FFFFFF",
            "borderRadius": 10,
            "height": 45,
            "justifyContent": "center",
            "marginTop": 20,
            "opacity": 1,
            "shadowColor": "#000000",
            "shadowOffset": Object {
              "height": 6,
              "width": 0,
            },
            "shadowOpacity": 0.33,
            "shadowRadius": 7.5,
            "width": 215,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#000000",
              "fontFamily": "System",
              "fontSize": 15,
              "textAlign": "center",
            }
          }
        >
          Merci !
        </Text>
      </View>
    </View>
  </View>
</Modal>
`;

exports[`DailyRewardModalComponent rendering Should render correctly the component 1`] = `
<Modal
  hardwareAccelerated={false}
  onRequestClose={[Function]}
  transparent={true}
  visible={true}
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#00000040",
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#ffffff",
            "borderRadius": 5,
            "width": "70%",
          },
          Object {
            "paddingBottom": 10,
          },
        ]
      }
    >
      <Text
        style={
          Object {
            "fontSize": 15,
            "fontWeight": "bold",
            "marginTop": 10,
            "textAlign": "center",
          }
        }
      >
        Récompense quotidienne
      </Text>
      <View
        style={
          Object {
            "borderBottomColor": "#000000",
            "borderBottomWidth": 1,
            "marginVertical": 10,
            "width": "50%",
          }
        }
      />
      <Text
        style={
          Object {
            "fontSize": 12,
            "fontWeight": "bold",
            "marginHorizontal": 10,
            "textAlign": "center",
          }
        }
      >
        Vous vous êtes connecté à EveryBet undefined jours d'affilé !
      </Text>
      <Text
        style={
          Object {
            "fontSize": 12,
            "fontWeight": "bold",
            "marginHorizontal": 10,
            "textAlign": "center",
          }
        }
      >
        Vous avez gagné 0
        <Image
          source={
            Object {
              "testUri": "../../../assets/images/betcoin_icon.png",
            }
          }
          style={
            Object {
              "height": 12,
              "width": 12,
            }
          }
        />
         comme cadeau pour votre fidélité.
      </Text>
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignSelf": "center",
            "backgroundColor": "#FFFFFF",
            "borderRadius": 10,
            "height": 45,
            "justifyContent": "center",
            "marginTop": 20,
            "opacity": 1,
            "shadowColor": "#000000",
            "shadowOffset": Object {
              "height": 6,
              "width": 0,
            },
            "shadowOpacity": 0.33,
            "shadowRadius": 7.5,
            "width": 215,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#000000",
              "fontFamily": "System",
              "fontSize": 15,
              "textAlign": "center",
            }
          }
        >
          Merci !
        </Text>
      </View>
    </View>
  </View>
</Modal>
`;
